### PR TITLE
Refactor/sqdsdks 6565 identity api client send identity request rob

### PR DESCRIPTION
**Pull request by rmi22186**
## Instructions
 1. PR target branch should be against `development`
 2. PR title name should follow this format: https://github.com/mParticle/mparticle-workflows/blob/main/.github/workflows/pr-title-check.yml
 3. PR branch prefix should follow this format: https://github.com/mParticle/mparticle-workflows/blob/main/.github/workflows/pr-branch-check-name.yml

 ## Summary
Fix integration tests due to XHR --> Fetch 

 ## Testing Plan
 - [X] Was this tested locally? If not, explain why. - Unit tests fix up

 ## Reference Issue (For mParticle employees only.  Ignore if you are an outside contributor)
 - Closes https://go.mparticle.com/work/REPLACEME

### CODE REVIEW

File: test/src/config/utils.js [hunk 1]
```diff
@@ -603,8 +608,22 @@ var pluses = /\+/g,
                 }
             })();
         });
+    },
+    fetchMockSuccess = function (url, body) {
```
Comment (alexs-mparticle) [inline test/src/config/utils.js:612]: Add the Cleanup Ticket here so we can search for it

File: test/src/config/utils.js [hunk 2]
```diff
@@ -603,8 +608,22 @@ var pluses = /\+/g,
...
+        fetchMock.post(
+            url,
+            {
+                status: 200,
+                body: JSON.stringify(body),
+            },
+            { overwriteRoutes: true }
+        );
+    },
```
Comment (alexs-mparticle) [inline test/src/config/utils.js:621]: Add the Cleanup Ticket here so we can search for it

File: test/src/tests-consent.ts [hunk 3]
```diff
@@ -632,6 +628,8 @@ describe('Consent', function() {
     });
 
     it('should have CCPA in payload', done => {
+        debugger;
```
Comment (alexs-mparticle) [inline test/src/tests-consent.ts:631]: Remove this

File: test/src/tests-consent.ts [hunk 4]
```diff
@@ -642,22 +640,29 @@ describe('Consent', function() {
...
+            testEvent.consent_state.ccpa.should.have.property('data_sale_opt_out');
+            testEvent.consent_state.ccpa.data_sale_opt_out.should.have.property('consented', true);
+            testEvent.consent_state.ccpa.data_sale_opt_out.should.have.property('timestamp_unixtime_ms', timestamp);
+            testEvent.consent_state.ccpa.data_sale_opt_out.should.have.property('document', 'consentDoc');
+            testEvent.consent_state.ccpa.data_sale_opt_out.should.have.property('location', 'location');
+            testEvent.consent_state.ccpa.data_sale_opt_out.should.have.property('hardware_id', 'hardware');
+    
+            done();
+        })
+        // })
```
Comment (alexs-mparticle) [inline test/src/tests-consent.ts:664]: Remove

File: test/src/tests-consent.ts [hunk 5]
```diff
@@ -642,22 +640,29 @@ describe('Consent', function() {
...
+            testEvent.consent_state.should.have.property('ccpa');
+            testEvent.consent_state.ccpa.should.have.property('data_sale_opt_out');
+            testEvent.consent_state.ccpa.data_sale_opt_out.should.have.property('consented', true);
+            testEvent.consent_state.ccpa.data_sale_opt_out.should.have.property('timestamp_unixtime_ms', timestamp);
+            testEvent.consent_state.ccpa.data_sale_opt_out.should.have.property('document', 'consentDoc');
+            testEvent.consent_state.ccpa.data_sale_opt_out.should.have.property('location', 'location');
+            testEvent.consent_state.ccpa.data_sale_opt_out.should.have.property('hardware_id', 'hardware');
+    
+            done();
+        })
```
Comment (alexs-mparticle) [inline test/src/tests-consent.ts:663]: Add Catch

File: test/src/tests-consent.ts [hunk 6]
```diff
@@ -679,32 +684,36 @@ describe('Consent', function() {
...
+            testEvent.consent_state.should.have.property('gdpr');
+            testEvent.consent_state.gdpr.should.have.property('test purpose');
+            testEvent.consent_state.gdpr['test purpose'].should.have.property('consented', false);
+            testEvent.consent_state.gdpr['test purpose'].should.have.property('timestamp_unixtime_ms', timestamp);
+            testEvent.consent_state.gdpr['test purpose'].should.have.property('document', 'consentDoc');
+            testEvent.consent_state.gdpr['test purpose'].should.have.property('location', 'location');
+            testEvent.consent_state.gdpr['test purpose'].should.have.property('hardware_id', 'hardware');
+            
+            done();
+        })
```
Comment (alexs-mparticle) [inline test/src/tests-consent.ts:716]: Add Catch

File: test/src/tests-cookie-syncing.js [hunk 7]
```diff
@@ -64,7 +57,7 @@ describe('cookie syncing', function() {
 
     afterEach(function() {
         fetchMock.restore();
-        mockServer.restore();
+        // mockServer.restore();
```
Comment (alexs-mparticle) [inline test/src/tests-cookie-syncing.js:60]: Remove

File: test/src/tests-cookie-syncing.js [hunk 8]
```diff
@@ -73,15 +66,18 @@ describe('cookie syncing', function() {
...
+            setTimeout(function() {
+                Should(
+                    mParticle.getInstance()._Store.pixelConfigurations.length
+                ).equal(1);
+                const data = mParticle.getInstance()._Persistence.getLocalStorage();
+                data[testMPID].csd.should.have.property('5');
+                
+                done();
+            }, timeout);
+        })
```
Comment (alexs-mparticle) [inline test/src/tests-cookie-syncing.js:80]: Add Catch

File: test/src/tests-cookie-syncing.js [hunk 9]
```diff
@@ -119,54 +115,67 @@ describe('cookie syncing', function() {
...
 
-            const data = mParticle.getInstance()._Persistence.getLocalStorage();
-
-            data[testMPID].csd.should.have.property(
-                5,
-                mParticle.getInstance()._Persistence.getLocalStorage().testMPID
+        waitForCondition(hasIdentifyReturned)
+        .then(() =>  {
+            setTimeout(function() {
+                // mockServer.requests = [];
```
Comment (alexs-mparticle) [inline test/src/tests-cookie-syncing.js:122]: Remove

File: test/src/tests-cookie-syncing.js [hunk 10]
```diff
@@ -119,54 +115,67 @@ describe('cookie syncing', function() {
...
-            mockServer.requests = [];
 
-            const data = mParticle.getInstance()._Persistence.getLocalStorage();
-
-            data[testMPID].csd.should.have.property(
-                5,
-                mParticle.getInstance()._Persistence.getLocalStorage().testMPID
+        waitForCondition(hasIdentifyReturned)
+        .then(() =>  {
+            setTimeout(function() {
```
Comment (alexs-mparticle) [inline test/src/tests-cookie-syncing.js:121]: Let's leave a note to explain why we're doing a setTimeout

File: test/src/tests-cookie-syncing.js [hunk 11]
```diff
@@ -119,54 +115,67 @@ describe('cookie syncing', function() {
...
-            done();
-        }, timeout);
+                );
+                Should(
+                    mParticle.getInstance()._Store.pixelConfigurations.length
+                ).equal(1);
+                
+                done();
+            }, timeout);
+        })
```
Comment (alexs-mparticle) [inline test/src/tests-cookie-syncing.js:137]: Add catch

File: test/src/tests-cookie-syncing.js [hunk 12]
```diff
@@ -119,54 +115,67 @@ describe('cookie syncing', function() {
...
+                        Should(
+                            mParticle.getInstance()._Store.pixelConfigurations.length
+                        ).equal(1);
+                        
+                        done();
+                    }, timeout);
+                })
             }, timeout);
-        }, timeout);
+        })
```
Comment (alexs-mparticle) [inline test/src/tests-cookie-syncing.js:178]: Add catch

File: test/src/tests-cookie-syncing.js [hunk 13]
```diff
@@ -119,54 +115,67 @@ describe('cookie syncing', function() {
...
-                    .getInstance()
-                    ._Persistence.getLocalStorage();
-                data1[testMPID].csd[5].should.be.ok();
-                data2['otherMPID'].csd[5].should.be.ok();
-                Should(
-                    mParticle.getInstance()._Store.pixelConfigurations.length
-                ).equal(1);
-
-                done();
+                const data1 = mParticle
```
Comment (alexs-mparticle) [inline test/src/tests-cookie-syncing.js:149]: I think we should add a note to explain why we're doing a setTimeout

File: test/src/tests-cookie-syncing.js [hunk 14]
```diff
@@ -292,37 +301,50 @@ describe('cookie syncing', function() {
...
-            }, timeout);
-        }, timeout);
+        waitForCondition(hasIdentifyReturned)
+        .then(() =>  {
+            // add pixels to preInitConfig
+            mParticle.init(apiKey, window.mParticle.config);
+            
+            waitForCondition(() => {
+                return (
+                    mParticle.getInstance()._Store.configurationLoaded === true
```
Comment (alexs-mparticle) [inline test/src/tests-cookie-syncing.js:311]: Should we make this a utility function?

File: test/src/tests-cookie-syncing.js [hunk 15]
```diff
@@ -292,37 +301,50 @@ describe('cookie syncing', function() {
...
+                );
+            })
+            .then(() => {
+                setTimeout(function() {
```
Comment (alexs-mparticle) [inline test/src/tests-cookie-syncing.js:315]: Let's explain why we're doing a set timeout

File: test/src/tests-cookie-syncing.js [hunk 16]
```diff
@@ -292,37 +301,50 @@ describe('cookie syncing', function() {
...
+                                Object.keys(cookies['MPID1'].csd).length.should.equal(1);
+                                
+                                done();
+                            }, timeout);
+                            
+                        })
+                    
+                }, timeout);
+            })
+        })
```
Comment (alexs-mparticle) [inline test/src/tests-cookie-syncing.js:345]: Add catch

File: test/src/tests-cookie-syncing.js [hunk 17]
```diff
@@ -982,70 +1004,77 @@ describe('cookie syncing', function() {
...
+                            newLocalStorage.testMPID.should.have.property('csd');
+                            newLocalStorage.testMPID.csd.should.have.property(5);
+                            done();
+                        }, timeout);
+                    }, timeout);
                 }, timeout);
-            }, timeout);
-        }, timeout);
+            })
+        })
```
Comment (alexs-mparticle) [inline test/src/tests-cookie-syncing.js:1077]: Add catch

File: test/src/tests-cookie-syncing.js [hunk 18]
```diff
@@ -1118,69 +1147,77 @@ describe('cookie syncing', function() {
...
+                            newLocalStorage.testMPID.csd.should.have.property(5);
+                            
+                            done();
+                        }, timeout);
+                    }, timeout);
                 }, timeout);
-            }, timeout);
-        }, timeout);
+            })
+        })
```
Comment (alexs-mparticle) [inline test/src/tests-cookie-syncing.js:1220]: Add catch

**Review by alexs-mparticle — CHANGES_REQUESTED** [review]

**Comment by rmi22186** [comment]
closing in favor of https://github.com/mParticle/mparticle-web-sdk/pull/930

**Pull request closed without merging**